### ISSUE TITLE
Missing extraction parameters

### ISSUE THREAD

**Issue by stscieisenhamer** [spacetelescope/jwst#2745]
#  Issue

From WFR2-OTB-20181004 processing run doggett/tests/wfr2/cstring/JWSTDP-2018_2-181017, running `calwebb_nrslamp-spec2.cfg`, gives the following:

<details><summary>Show log...</summary>
<p>

```
$ strun /ifs/int/jwstc/pipeline/pkgs/miniconda3/envs/jwstdp-0.12.0-py36/lib/python3.6/site-packages/jwst-0.12.0-py3.6-linux-x86_64.egg/jwst/pipeline//calwebb_nrslamp-spec2.cfg /ifs/int/jwstc/store/doggett/tests/wfr2/cstring/JWSTDP-2018_2-181017/2018-10-19-215606/sdp/asn_creation/cal/level2/jw00791-o004_20181019t221608_nrslamp-spec2_002_asn.json
2018-10-22 15:53:21,247 - stpipe.Spec2NRSLamp - INFO - Spec2Pipeline instance created.
2018-10-22 15:53:21,248 - stpipe.Spec2NRSLamp.bkg_subtract - INFO - BackgroundStep instance created.
2018-10-22 15:53:21,249 - stpipe.Spec2NRSLamp.assign_wcs - INFO - AssignWcsStep instance created.
2018-10-22 15:53:21,250 - stpipe.Spec2NRSLamp.imprint_subtract - INFO - ImprintStep instance created.
2018-10-22 15:53:21,251 - stpipe.Spec2NRSLamp.msa_flagging - INFO - MSAFlagOpenStep instance created.
2018-10-22 15:53:21,252 - stpipe.Spec2NRSLamp.extract_2d - INFO - Extract2dStep instance created.
2018-10-22 15:53:21,253 - stpipe.Spec2NRSLamp.flat_field - INFO - FlatFieldStep instance created.
2018-10-22 15:53:21,254 - stpipe.Spec2NRSLamp.srctype - INFO - SourceTypeStep instance created.
2018-10-22 15:53:21,255 - stpipe.Spec2NRSLamp.straylight - INFO - StraylightStep instance created.
2018-10-22 15:53:21,256 - stpipe.Spec2NRSLamp.fringe - INFO - FringeStep instance created.
2018-10-22 15:53:21,257 - stpipe.Spec2NRSLamp.pathloss - INFO - PathLossStep instance created.
2018-10-22 15:53:21,290 - stpipe.Spec2NRSLamp.barshadow - INFO - BarShadowStep instance created.
2018-10-22 15:53:21,291 - stpipe.Spec2NRSLamp.photom - INFO - PhotomStep instance created.
2018-10-22 15:53:21,293 - stpipe.Spec2NRSLamp.resample_spec - INFO - ResampleSpecStep instance created.
2018-10-22 15:53:21,295 - stpipe.Spec2NRSLamp.cube_build - INFO - CubeBuildStep instance created.
2018-10-22 15:53:21,296 - stpipe.Spec2NRSLamp.extract_1d - INFO - Extract1dStep instance created.
2018-10-22 15:53:21,311 - stpipe - INFO - Hostname: iljwdmscdp1.stsci.edu
2018-10-22 15:53:21,311 - stpipe - INFO - OS: Linux
2018-10-22 15:53:21,332 - stpipe.Spec2NRSLamp - INFO - Step Spec2NRSLamp running with args ('/ifs/int/jwstc/store/doggett/tests/wfr2/cstring/JWSTDP-2018_2-181017/2018-10-19-215606/sdp/asn_creation/cal/level2/jw00791-o004_20181019t221608_nrslamp-spec2_002_asn.json',).
2018-10-22 15:53:21,935 - stpipe.Spec2NRSLamp - INFO - Prefetching reference files for dataset: 'jw00791004001_02101_00001_nrs2_rate.fits' reftypes = ['barshadow', 'camera', 'collimator', 'cubepar', 'disperser', 'distortion', 'drizpars', 'extract1d', 'filteroffset', 'fore', 'fpa', 'ifufore', 'ifupost', 'ifuslicer', 'msa', 'msaoper', 'ote', 'regions', 'resol', 'specwcs', 'wavecorr', 'wavelengthrange', 'wfssbkg']
2018-10-22 15:53:22,320 - stpipe.Spec2NRSLamp - INFO - Prefetch for BARSHADOW reference file is 'N/A'.
2018-10-22 15:53:22,321 - stpipe.Spec2NRSLamp - INFO - Prefetch for CAMERA reference file is '/grp/crds/cache/references/jwst/jwst_nirspec_camera_0004.asdf'.
2018-10-22 15:53:22,322 - stpipe.Spec2NRSLamp - INFO - Prefetch for COLLIMATOR reference file is '/grp/crds/cache/references/jwst/jwst_nirspec_collimator_0004.asdf'.
2018-10-22 15:53:22,326 - stpipe.Spec2NRSLamp - INFO - Prefetch for CUBEPAR reference file is 'N/A'.
2018-10-22 15:53:22,326 - stpipe.Spec2NRSLamp - INFO - Prefetch for DISPERSER reference file is '/grp/crds/cache/references/jwst/jwst_nirspec_disperser_0038.asdf'.
2018-10-22 15:53:22,330 - stpipe.Spec2NRSLamp - INFO - Prefetch for DISTORTION reference file is 'N/A'.
2018-10-22 15:53:22,330 - stpipe.Spec2NRSLamp - INFO - Prefetch for DRIZPARS reference file is 'N/A'.
2018-10-22 15:53:22,330 - stpipe.Spec2NRSLamp - INFO - Prefetch for EXTRACT1D reference file is '/grp/crds/cache/references/jwst/jwst_nirspec_extract1d_0004.json'.
2018-10-22 15:53:22,332 - stpipe.Spec2NRSLamp - INFO - Prefetch for FILTEROFFSET reference file is 'N/A'.
2018-10-22 15:53:22,332 - stpipe.Spec2NRSLamp - INFO - Prefetch for FORE reference file is 'N/A'.
2018-10-22 15:53:22,332 - stpipe.Spec2NRSLamp - INFO - Prefetch for FPA reference file is '/grp/crds/cache/references/jwst/jwst_nirspec_fpa_0005.asdf'.
2018-10-22 15:53:22,335 - stpipe.Spec2NRSLamp - INFO - Prefetch for IFUFORE reference file is 'N/A'.
2018-10-22 15:53:22,335 - stpipe.Spec2NRSLamp - INFO - Prefetch for IFUPOST reference file is 'N/A'.
2018-10-22 15:53:22,335 - stpipe.Spec2NRSLamp - INFO - Prefetch for IFUSLICER reference file is 'N/A'.
2018-10-22 15:53:22,335 - stpipe.Spec2NRSLamp - INFO - Prefetch for MSA reference file is '/grp/crds/cache/references/jwst/jwst_nirspec_msa_0005.asdf'.
2018-10-22 15:53:22,337 - stpipe.Spec2NRSLamp - INFO - Prefetch for MSAOPER reference file is '/grp/crds/cache/references/jwst/jwst_nirspec_msaoper_0001.json'.
2018-10-22 15:53:22,340 - stpipe.Spec2NRSLamp - INFO - Prefetch for OTE reference file is '/grp/crds/cache/references/jwst/jwst_nirspec_ote_0005.asdf'.
2018-10-22 15:53:22,341 - stpipe.Spec2NRSLamp - INFO - Prefetch for REGIONS reference file is 'N/A'.
2018-10-22 15:53:22,341 - stpipe.Spec2NRSLamp - INFO - Prefetch for RESOL reference file is 'N/A'.
2018-10-22 15:53:22,341 - stpipe.Spec2NRSLamp - INFO - Prefetch for SPECWCS reference file is 'N/A'.
2018-10-22 15:53:22,341 - stpipe.Spec2NRSLamp - INFO - Prefetch for WAVECORR reference file is 'N/A'.
2018-10-22 15:53:22,341 - stpipe.Spec2NRSLamp - INFO - Prefetch for WAVELENGTHRANGE reference file is '/grp/crds/cache/references/jwst/jwst_nirspec_wavelengthrange_0004.asdf'.
2018-10-22 15:53:22,343 - stpipe.Spec2NRSLamp - INFO - Prefetch for WFSSBKG reference file is 'N/A'.
2018-10-22 15:53:22,343 - stpipe.Spec2NRSLamp - INFO - Starting calwebb_spec2 ...
2018-10-22 15:53:22,353 - stpipe.Spec2NRSLamp - INFO - Processing product jw00791004001_02101_00001_nrs2
2018-10-22 15:53:22,353 - stpipe.Spec2NRSLamp - INFO - Working on input /ifs/int/jwstc/store/doggett/tests/wfr2/cstring/JWSTDP-2018_2-181017/2018-10-19-215606/sdp/asn_creation/cal/level2/jw00791004001_02101_00001_nrs2_rate.fits ...
2018-10-22 15:53:22,666 - stpipe.Spec2NRSLamp.assign_wcs - INFO - Step assign_wcs running with args (<ImageModel(2048, 2048) from jw00791004001_02101_00001_nrs2_rate.fits>,).
2018-10-22 15:53:22,787 - stpipe.Spec2NRSLamp.assign_wcs - INFO - gwa_ytilt is 0.1973863540000002 deg
2018-10-22 15:53:22,787 - stpipe.Spec2NRSLamp.assign_wcs - INFO - gwa_xtilt is 0.3432061670000008 deg
2018-10-22 15:53:22,788 - stpipe.Spec2NRSLamp.assign_wcs - INFO - theta_y correction: 0.0027134787700957316 deg
2018-10-22 15:53:22,789 - stpipe.Spec2NRSLamp.assign_wcs - INFO - theta_x correction: 0.00022852898916941595 deg
2018-10-22 15:53:23,405 - stpipe.Spec2NRSLamp.assign_wcs - INFO - Removing slit S200A1 from the list of open slits because the WCS bounding_box is completely outside the detector.
2018-10-22 15:53:23,475 - stpipe.Spec2NRSLamp.assign_wcs - INFO - Removing slit S200A2 from the list of open slits because the WCS bounding_box is completely outside the detector.
2018-10-22 15:53:23,544 - stpipe.Spec2NRSLamp.assign_wcs - INFO - Removing slit S400A1 from the list of open slits because the WCS bounding_box is completely outside the detector.
2018-10-22 15:53:23,626 - stpipe.Spec2NRSLamp.assign_wcs - INFO - Removing slit S1600A1 from the list of open slits because the WCS bounding_box is completely outside the detector.
2018-10-22 15:53:23,700 - stpipe.Spec2NRSLamp.assign_wcs - INFO - Slits projected on detector NRS2: ['S200B1']
2018-10-22 15:53:23,700 - stpipe.Spec2NRSLamp.assign_wcs - INFO - Computing WCS for 1 open slitlets
2018-10-22 15:53:23,729 - stpipe.Spec2NRSLamp.assign_wcs - INFO - gwa_ytilt is 0.1973863540000002 deg
2018-10-22 15:53:23,729 - stpipe.Spec2NRSLamp.assign_wcs - INFO - gwa_xtilt is 0.3432061670000008 deg
2018-10-22 15:53:23,730 - stpipe.Spec2NRSLamp.assign_wcs - INFO - theta_y correction: 0.0027134787700957316 deg
2018-10-22 15:53:23,731 - stpipe.Spec2NRSLamp.assign_wcs - INFO - theta_x correction: 0.00022852898916941595 deg
2018-10-22 15:53:23,746 - stpipe.Spec2NRSLamp.assign_wcs - INFO - SPORDER= -1, wrange=[6e-07, 5.3e-06]
2018-10-22 15:53:24,188 - stpipe.Spec2NRSLamp.assign_wcs - INFO - There are 0 open slits in quadrant 1
2018-10-22 15:53:24,188 - stpipe.Spec2NRSLamp.assign_wcs - INFO - There are 0 open slits in quadrant 2
2018-10-22 15:53:24,188 - stpipe.Spec2NRSLamp.assign_wcs - INFO - There are 0 open slits in quadrant 3
2018-10-22 15:53:24,188 - stpipe.Spec2NRSLamp.assign_wcs - INFO - There are 0 open slits in quadrant 4
2018-10-22 15:53:24,189 - stpipe.Spec2NRSLamp.assign_wcs - INFO - There are 1 open slits in quadrant 5
2018-10-22 15:53:24,379 - stpipe.Spec2NRSLamp.assign_wcs - INFO - Created a NIRSPEC nrs_lamp pipeline with references {'distortion': None, 'filteroffset': None, 'specwcs': None, 'regions': None, 'wavelengthrange': '/grp/crds/cache/references/jwst/jwst_nirspec_wavelengthrange_0004.asdf', 'camera': '/grp/crds/cache/references/jwst/jwst_nirspec_camera_0004.asdf', 'collimator': '/grp/crds/cache/references/jwst/jwst_nirspec_collimator_0004.asdf', 'disperser': '/grp/crds/cache/references/jwst/jwst_nirspec_disperser_0038.asdf', 'fore': None, 'fpa': '/grp/crds/cache/references/jwst/jwst_nirspec_fpa_0005.asdf', 'msa': '/grp/crds/cache/references/jwst/jwst_nirspec_msa_0005.asdf', 'ote': '/grp/crds/cache/references/jwst/jwst_nirspec_ote_0005.asdf', 'ifupost': None, 'ifufore': None, 'ifuslicer': None}
2018-10-22 15:53:24,522 - stpipe.Spec2NRSLamp.assign_wcs - INFO - COMPLETED assign_wcs
2018-10-22 15:53:24,530 - stpipe.Spec2NRSLamp.assign_wcs - INFO - Step assign_wcs done
2018-10-22 15:53:24,584 - stpipe.Spec2NRSLamp.flat_field - INFO - Step flat_field running with args (<ImageModel(2048, 2048) from jw00791004001_02101_00001_nrs2_rate.fits>,).
2018-10-22 15:53:24,584 - stpipe.Spec2NRSLamp.flat_field - INFO - Step skipped.
2018-10-22 15:53:24,585 - stpipe.Spec2NRSLamp.flat_field - INFO - Step flat_field done
2018-10-22 15:53:24,620 - stpipe.Spec2NRSLamp.srctype - INFO - Step srctype running with args (<ImageModel(2048, 2048) from jw00791004001_02101_00001_nrs2_rate.fits>,).
2018-10-22 15:53:24,635 - stpipe.Spec2NRSLamp.srctype - INFO - Input EXP_TYPE is NRS_LAMP
2018-10-22 15:53:24,636 - stpipe.Spec2NRSLamp.srctype - WARNING - EXP_TYPE NRS_LAMP not applicable to this operation
2018-10-22 15:53:24,636 - stpipe.Spec2NRSLamp.srctype - WARNING - Step will be skipped
2018-10-22 15:53:24,637 - stpipe.Spec2NRSLamp.srctype - INFO - Step srctype done
2018-10-22 15:53:24,670 - stpipe.Spec2NRSLamp.photom - INFO - Step photom running with args (<ImageModel(2048, 2048) from jw00791004001_02101_00001_nrs2_rate.fits>,).
2018-10-22 15:53:24,670 - stpipe.Spec2NRSLamp.photom - INFO - Step skipped.
2018-10-22 15:53:24,671 - stpipe.Spec2NRSLamp.photom - INFO - Step photom done
2018-10-22 15:53:24,705 - stpipe.Spec2NRSLamp.extract_1d - INFO - Step extract_1d running with args (<ImageModel(2048, 2048) from jw00791004001_02101_00001_nrs2_cal.fits>,).
2018-10-22 15:53:24,728 - stpipe.Spec2NRSLamp.extract_1d - INFO - Using EXTRACT1D reference file /grp/crds/cache/references/jwst/jwst_nirspec_extract1d_0004.json
2018-10-22 15:53:24,758 - stpipe.Spec2NRSLamp.extract_1d - CRITICAL - Missing extraction parameters.
Traceback (most recent call last):
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 91, in process
    asn.filename
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 302, in process_exposure_product
    x1d_result = self.extract_1d(result_extra)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/extract_1d/extract_1d_step.py", line 117, in process
    self.log_increment)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/extract_1d/extract.py", line 2189, in do_extract1d
    raise ValueError('Missing extraction parameters.')
ValueError: Missing extraction parameters.
----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec2Pipeline':
    One or more products failed to process. Failing calibration.
Traceback (most recent call last):
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 107, in process
    'One or more products failed to process. Failing calibration.'
RuntimeError: One or more products failed to process. Failing calibration.

----------------------------------------------------------------------
Traceback (most recent call last):
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/EGG-INFO/scripts/strun", line 26, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/step.py", line 162, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/ifs/int/jwstc/info/eisenham/anaconda3/envs/condadev/lib/python3.6/site-packages/jwst-0.13.0a0.dev67+gf56c3fd4-py3.6-linux-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 107, in process
    'One or more products failed to process. Failing calibration.'
RuntimeError: One or more products failed to process. Failing calibration.
```

</p>
</details>

**Comment by stscieisenhamer** [comment]
# Resources

Data can be found in `/grp/jwst/ssb/eisenhamer/jwst_issues/issue2745`.

**Comment by philhodge** [comment]
The input FITS file jw00791004001_02101_00001_nrs2_rate.fits has

EXP_TYPE= 'NRS_LAMP'
FILTER  = 'OPAQUE  '
GRATING = 'MIRROR  '

But the extract_1d reference file jwst_nirspec_extract1d_0004.json has "exp_type": "NRS_FIXEDSLIT".  Perhaps we need to adjust the selection criteria.

**Comment by hbushouse** [comment]
The processing of NRS_LAMP exposures was supposedly handled in #1782, but from what I can tell the only updates that were ever made were to handle EXP_TYPE='NRS_LAMP' and/or FILTER='OPAQUE' in the association rules (#2177) and the creation of the new `calwebb_spec2-nrslamp.cfg` in order to skip certain pipeline steps. But it was also mentioned in #1782 that these types of exposures needed to be handled in a special way in the `assign_wcs` step. Was that ever implemented? Furthermore, did anyone ever try testing the `calwebb_spec2-nrslamp.cfg` pipeline on one of these exposures? If so, then we should've uncovered the `extract_1d` problem sooner.

At the very least, we now probably need to add EXP_TYPE='NRS_LAMP' to the extract1d reference file. But we also need to know which of the 5 possible slits/apertures to extract (all of them?).

**Comment by hbushouse** [comment]
Wait a sec ... GRATING='MIRROR'. I'm pretty sure that means no dispersive element was in the optical path and hence the data are imaging-like. So there's no spectra to extract. Looking at the image tends to confirm this:
![rate](https://user-images.githubusercontent.com/12860084/47384318-c6b59580-d6d4-11e8-8b64-f10e7a3b4805.jpeg)

**Comment by hbushouse** [comment]
@nden points out that this also requires a change to `assign_wcs` to handle GRATING='MIRROR' as an imaging mode.

**Comment by nden** [comment]
The part of this issue related to assign_wcs was resolved in #2761 .
